logical/plugin/logger.go

```diff
@@ -0,0 +1,238 @@
+package plugin
+
+import (
+	"fmt"
+	"net/rpc"
+
+	plugin "github.com/hashicorp/go-plugin"
+	log "github.com/mgutz/logxi/v1"
+)
+
+type LoggerClient struct {
+	client *rpc.Client
+}
+
+func (l *LoggerClient) Trace(msg string, args ...interface{}) {
+	cArgs := &LoggerArgs{
+		Msg:  msg,
+		Args: args,
+	}
+	l.client.Call("Plugin.Trace", cArgs, &struct{}{})
+}
+
+func (l *LoggerClient) Debug(msg string, args ...interface{}) {
+	cArgs := &LoggerArgs{
+		Msg:  msg,
+		Args: args,
+	}
+	l.client.Call("Plugin.Debug", cArgs, &struct{}{})
+}
+
+func (l *LoggerClient) Info(msg string, args ...interface{}) {
+	cArgs := &LoggerArgs{
+		Msg:  msg,
+		Args: args,
+	}
+	l.client.Call("Plugin.Info", cArgs, &struct{}{})
+}
+func (l *LoggerClient) Warn(msg string, args ...interface{}) error {
+	var reply LoggerReply
+	cArgs := &LoggerArgs{
+		Msg:  msg,
+		Args: args,
+	}
+	err := l.client.Call("Plugin.Warn", cArgs, &reply)
+	if err != nil {
+		return err
+	}
+	if reply.Error != nil {
+		return reply.Error
+	}
+
+	return nil
+}
+func (l *LoggerClient) Error(msg string, args ...interface{}) error {
+	var reply LoggerReply
+	cArgs := &LoggerArgs{
+		Msg:  msg,
+		Args: args,
+	}
+	err := l.client.Call("Plugin.Error", cArgs, &reply)
+	if err != nil {
+		return err
+	}
+	if reply.Error != nil {
+		return reply.Error
+	}
+
+	return nil
+}
+
+func (l *LoggerClient) Fatal(msg string, args ...interface{}) {
+	var reply LoggerReply
+	cArgs := &LoggerArgs{
+		Msg:  msg,
+		Args: args,
+	}
+	l.client.Call("Plugin.Fatal", cArgs, &reply)
+
+	// Panic on the client side. In this case the panic would be on the
+	// backend plugin which could be restarted, as opposed to panicking
+	// from vault client.
+	if reply.Error != nil {
+		panicMsg := fmt.Sprintf("Exit due to fatal error: %s", reply.Error)
+		panic(panicMsg)
```

Comment:
This is the concrete client implementation of `logxi`'s `Logger` interface. I can have `Fatal()` be a no-op, but I thought I'd implement it and perform the RPC call just in case one day we do use `log.Fatal()`. Since that call panics on the server (which is on vault side, not plugin side), I went with recovering and then panicking on the client instead.